base-lite/ub/ub_test.go

```diff
@@ -0,0 +1,329 @@
+package main
+
+import (
+	"os"
+	"reflect"
+	"testing"
+)
+
+func assertEqual(a string, b string, t *testing.T) {
+	if a != b {
+		t.Error(a + " != " + b)
+	}
+}
+
+func Test_ensure(t *testing.T) {
+	type args struct {
+		envVar string
+	}
+	err := os.Setenv("ENV_VAR", "value")
+	if err != nil {
+		t.Error("Unable to set ENV_VAR for the test")
+	}
+	tests := []struct {
+		name string
+		args args
+		want bool
+	}{
+		{
+			name: "should exist",
+			args: args{
+				envVar: "ENV_VAR",
+			},
+			want: true,
+		},
+		{
+			name: "should not exist",
+			args: args{
+				envVar: "RANDOM_ENV_VAR",
+			},
+			want: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := ensure(tt.args.envVar); got != tt.want {
+				t.Errorf("ensure() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func Test_path(t *testing.T) {
+	type args struct {
+		filePath  string
+		operation string
+	}
+	const (
+		sampleFile       = "testResources/sampleFile"
+		sampleFile2      = "testResources/sampleFile2"
+		fileDoesNotExist = "testResources/sampleFile3"
+	)
+	err := os.Chmod(sampleFile, 0777)
+	if err != nil {
+		t.Error("Unable to set permissions for the file")
+	}
+	err = os.Chmod(sampleFile2, 0000)
+	if err != nil {
+		t.Error("Unable to set permissions for the file")
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    bool
+		wantErr bool
+	}{
+		{name: "file readable",
```

Comment:
nit: formatting can be improved here by changing it to
               `{
			name: "file readable",
			args: args{
				filePath: sampleFile,
				operation: "readable",
			},
			want:    true,
			wantErr: false,
		},`